### DISCUSSION TITLE
CPM and Ubuntu 18.04 LTS PPA builds

### DISCUSSION THREAD

**Discussion by christianparpart** [cpm-cmake/CPM.cmake#279] [Q&A] [answered]
Hey,

I sadly realized that CPM has a hard requirement on CMake >= 3.14. In my project, I am also maintaining a PPA (via launchpad) for convenience of installation and updating. This includes Ubuntu 18.04 LTS. But this still supported LTS version comes with CMake 3.10.2 and my build breaks because of that. 

As far as I am aware of, you can only specify requirements the PPA buildbot knows about, that is. what Ubuntu package system provides. Is there a way to either lift the requirements (by reducing non-necessary features, possibly) or getting it into a mode where it would still run on such systems?

**Comment by iboB** [comment, marked as answer]
Quite late with the response, but it's not possible I'm afraid. 

Or, rather, *waaay* outside of the scope of this project, CPM makes clever usage of huge features added in CMake 3.14. While they could theoretically be reimplemented for older versions of CMake, this is, first, a huge undertaking, and second, quite the sad experience for the developers that will have to create something which already exists, and is doomed to be deleted once the rest of the world moves to CMake 3.14+ (which is not that far into the future).
